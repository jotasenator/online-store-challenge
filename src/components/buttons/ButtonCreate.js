import React from 'react'
import { Link } from 'react-router-dom'

export default function ButtonCreate() {


    return (
        <div>
            <button
                className='btn btn-primary float-right'>
                <Link to={'/create'}>Create</Link>
            </button>
        </div>
    )
}
