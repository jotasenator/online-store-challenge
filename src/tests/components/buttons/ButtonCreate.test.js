import React from 'react'
import '@testing-library/jest-dom'
import { shallow } from 'enzyme'

import ButtonCreate from '../../../components/buttons/ButtonCreate'


describe('Test from ButtonCreate', () => {
    let wrapper;

    beforeEach(() => {
        wrapper = shallow(<ButtonCreate />);
    });

    afterEach(() => {
        wrapper.unmount();
    });

    test('should render <ButtonCreate/> correctly', () => {
        expect(wrapper).toMatchSnapshot()
    })

    test('should show what text is inside button', () => {
        expect(wrapper.find('button[className="btn btn-primary float-right"]').text()).toBe('Create')
        expect(wrapper.find('button').text()).toBe('Create')
        expect((wrapper.find('Link')).text()).toBe('Create')

    })

})
