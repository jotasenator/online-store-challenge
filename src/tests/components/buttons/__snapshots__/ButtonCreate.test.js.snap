// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test from ButtonCreate should render <ButtonCreate/> correctly 1`] = `
<div>
  <button
    className="btn btn-primary float-right"
  >
    <BrowserRouter>
      <Link
        to="/create"
      >
        Create
      </Link>
    </BrowserRouter>
  </button>
</div>
`;
